File: app/src/main/java/com/gasparaiciukas/owntrainer/network/GetService.java
package com.gasparaiciukas.owntrainer.network;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.Query;

public interface GetService {
    String BASE_URL = "https://api.edamam.com/api/food-database/v2/";

    @Headers({
            "Content-Type: application/json"
    })
    @GET("parser")
    Call<GetResponse> getResponse(@Query("app_id") String appId, @Query("app_key") String appKey,
                                  @Query("ingr") String query);
}

File: app/src/main/java/com/gasparaiciukas/owntrainer/network/GetService.kt
package com.gasparaiciukas.owntrainer.network

import com.gasparaiciukas.owntrainer.BuildConfig
import retrofit2.Response

interface GetService {
    suspend fun getFoods(
        apiKey: String = BuildConfig.API_KEY,
        query: String,
        dataType: String? = null,
        numberOfResultsPerPage: Int? = null,
        pageSize: Int? = null,
        pageNumber: Int? = null,
        sortBy: String? = null,
        sortOrder: String? = null,
        requireAllWords: Boolean? = null,
    ): Response<GetResponse>
}
